core/src/main/java/aima/core/util/datastructure/Tuple.java

```diff
@@ -0,0 +1,82 @@
+package aima.core.util.datastructure;
+
+/**
+ * Created by @AdrianBZG on 2/04/17.
+ */
+
+import java.util.function.Function;
+
+/**
+ * Util class: a parametric tuple type
+ */
+public class Tuple<A, B> {
+    private A fst;
```

Comment:
I'd write `first` instead of `fst`. The abbreviation isn't helping anyone.
see: clean code http://www.itiseezee.com/?p=83